In [19]:
import requests
import pandas as pd
import re
from bs4 import BeautifulSoup

Creating the master Dataframe -->

In [20]:
columns = ['Postal Code','Borough','Neighborhood']
Master_DF = pd.DataFrame(columns=columns)

I'll be using Beautiful Soup to parse the HTML content of the webpage

In [21]:
response = requests.get('https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M')
data = BeautifulSoup(response.text,'html.parser')

In [23]:
table = data.find_all('table')[0]
rows = table.find_all('tr')
print(len(rows))
print(rows[0])

181
<tr>
<th>Postal Code
</th>
<th>Borough
</th>
<th>Neighbourhood
</th></tr>


After inspecting the webpage, the required table is contained within the table tag, which I've extracted using the find_all command

Within the table , the individual rows are contained within the tr tag. The length of rows is 181 which means we have 180 records. The first record doesn't contin the information, it's just the name of columns

In [24]:
for x in rows[1:]:
    temp = x.find_all('td')
    list_temp = []
    for i in temp:
        list_temp.append(str(i.text.strip()))
    #print(list_temp)
    pc = list_temp[0]
    b = list_temp[1]
    n = list_temp[2]
    Master_DF = Master_DF.append({'Postal Code':pc,'Borough':b,'Neighborhood':n},ignore_index=True)

In [25]:
Master_DF.shape

(180, 3)

Iterating over the rows , the individual elements are contained within the <td> tag which I am accesing and appending to the Master Dataframe

In [26]:
Master_DF = Master_DF[Master_DF['Borough'] != 'Not assigned']
Master_DF.shape

(103, 3)